### PR TITLE
NEXT-39412 - Use common PHPStan config from core and update PHP tooling

### PR DESCRIPTION
**Pull request by mitelg**
See https://gitlab.shopware.com/shopware/6/product/platform/-/merge_requests/15444

### CODE REVIEW

**Review by ennasus4sun — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -35,5 +35,12 @@
         "psr-4": {
             "Swag\\PlatformDemoData\\": "src/"
         }
+    },
+    "scripts": {
+        "cs": "../../../vendor/bin/php-cs-fixer fix --dry-run",
+        "cs-fix": "../../../vendor/bin/php-cs-fixer fix",
```
Comment (MalteJanz) [inline composer.json:41]: Isn't it `ecs` and `ecs-fix` in platform and plugins?

**Review by MalteJanz — COMMENTED** [review]
Only have a small question regarding the naming of the composer command. Otherwise looks good to me and thanks for the contribution 💙

File: composer.json (hunk 1 above)
Comment (mitelg) [reply composer.json:41]: yeah, but due to "backward" compatibility. I would have re-named it everywhere, but people argued, that they have finger memories with "ecs-fix", so we should not break it. 

but here there was nothing before, so I added it with "cs*" (also on other plugins)

Comment (MalteJanz) [reply composer.json:41]: But that only makes it worse as we aren't consistent 😅 . So when I'm used to platform and just type `composer run ecs-fix` that wouldn't work with this specific plugin but it works e. g. in the Migration-Assistant plugin 🙈 

So please tell me you will fix it everywhere with the next major? or at least create tickets for every domain? 🤞 😁

Comment (mitelg) [reply composer.json:41]: mmmh 😬🙈😇 I will see, what I can do

**Pull request merged**